### ISSUE TITLE
add function to generate explicit schema mapping from polars datatype to pyarrow datatype

### ISSUE THREAD

**Issue by Sage0614** [pola-rs/polars#15563]
### Description

It would be helpful to have a function that explicitly create map from polars supported datatypes to pyarrow supported datatypes, this would make checks seirialization to disk, and verify external data source easier, as many other file format/ libraries has a defined mapping to pyarrow types.

ideally, the two helper functions are `to_arrow_schema(pl_schema:OrderedDict[str,pl.DataType]) -> pa.Schema` and `from_arrow_schema(pa_schema:pa.Schema)->OrderedDict[str,pl.DataType]`

**Comment by choucavalier** [comment]
```python
import polars as pl
import pyarrow as pa

pyarrow_schema = pa.schema([pa.field('hello', pa.string())])
pl.from_arrow(pyarrow_schema.empty_table()).schema # Schema({'hello': String})
```

**Comment by MarcoGorelli** [comment]
I don't know if this is in-scope for Polars (perhaps it is?) but in the meantime, Narwhals can translate schemas between libraries:
```python
In [1]: import narwhals as nw

In [2]: import pyarrow as pa

In [3]: pyarrow_schema = pa.schema([pa.field('hello', pa.string())])

In [4]: polars_schema = nw.Schema.from_arrow(pyarrow_schema).to_polars()

In [5]: polars_schema
Out[5]: Schema([('hello', String)])
```

**Comment by nameexhaustion** [comment]
For importing to polars, as of https://github.com/pola-rs/polars/pull/23879 you can pass a pyarrow schema directly - `pl.Schema(pyarrow_schema)`

**Comment by kylebarron** [comment]
I think this is closed now by https://github.com/pola-rs/polars/pull/24365?

**Comment by nameexhaustion** [comment]
> I think this is closed now by [#24365](https://github.com/pola-rs/polars/pull/24365)?

Not exactly, OP here likely wants a `to_arrow()` function on the `Schema` class itself.

**Comment by deanm0000** [comment]
> Not exactly, OP here likely wants a `to_arrow()` function on the `Schema` class itself.

I would like that but I just do `pl.DataFrame(schema={"a":pl.String}).to_arrow().schema` instead so it's not really a big deal.

**Comment by kylebarron** [comment]
`pa.schema(pl.Schema({"a":pl.String}))` is a bit simpler than that...

**Comment by deanm0000** [comment]
> `pa.schema(pl.Schema({"a":pl.String}))` is a bit simpler than that...

I'm not sure if you're saying that works already or just that it'd be a nice feature. If the former, I just tried it with pyarrow 22.0.0 and it doesn't work. It'd need pyarrow to recognize polars datatypes and auto convert them to pyarrow datatypes.

**Comment by kylebarron** [comment]
This already works today. It uses the pycapsule interface, the `__arrow_c_schema__` dunder method. It exports an Arrow schema that any other python library can access.

<img width="571" height="150" alt="Image" src="https://github.com/user-attachments/assets/a8ba8f5f-ef66-4015-9967-ce7de43f875a" />

I'm on polars 1.35.1 and pyarrow 22

This is what @nameexhaustion added in https://github.com/pola-rs/polars/pull/24365

**Comment by deanm0000** [comment]
just tried it again with 1.35.1 and it worked, I was in an old env with just 1.26. my bad.
